torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -701,9 +766,26 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
         # Switch to local shard after backward.
         self._use_param_local_shard([param])
 
+        # Prefetch previous layer's full params in backward pass post backward hook,
+        # If next layer's backward computation is done and full params are freed,
+        # no need to prefetch the full params again.
+        # Only prefetch full params if any of the next layer's outputs requires grad
+        if (
```

Comment:
nit: same comment about the helper function

Reply:
sounds good